app/screens/edit_profile/index.ts

```diff
@@ -22,9 +26,31 @@ const enhanced = withObservables([], ({database}: WithDatabaseArgs) => {
     const samlLastNameAttributeSet = observeConfigBooleanValue(database, 'SamlLastNameAttributeSet');
     const samlNicknameAttributeSet = observeConfigBooleanValue(database, 'SamlNicknameAttributeSet');
     const samlPositionAttributeSet = observeConfigBooleanValue(database, 'SamlPositionAttributeSet');
+    const customAttributesEnabled = observeConfigBooleanValue(database, 'FeatureFlagCustomProfileAttributes');
+
+    const rawCustomAttributes = currentUser.pipe(
+        switchMap((u) => (u ? observeCustomProfileAttributesByUserId(database, u.id) : of$([]))),
+    );
+    let formattedCustomAttributes;
+    let customFields;
+    if (customAttributesEnabled) {
+        customFields = observeCustomProfileFields(database);
+
+        // Convert attributes to the format expected by the component
+        formattedCustomAttributes = combineLatest([rawCustomAttributes, customFields]).pipe(
+            switchMap(([attributes]) => {
+                if (!attributes?.length) {
+                    return of$({} as CustomAttributeSet);
+                }
+
+                return from(convertProfileAttributesToCustomAttributes(database, attributes, sortCustomProfileAttributes));
```

Comment:
consider extracting the sort and stuff from this function to another function and then you won't need the promise here. That is basically cause `convertProfileAttributesToCustomAttributes` is querying the database again when the customFields you already have them